encoding/kzg/committer/doc.go

```diff
@@ -0,0 +1,78 @@
+// Package committer provides functions to create and verify EigenDA [encoding.BlobCommitments].
+//
+// Note that EigenDA blob commitments are not simply a single KZG commitment, but also
+// include the blob's length, as well as a proof of this length (LengthCommitment + LengthProof).
+// This complexity stems from the fact that EigenDA, unlike Ethereum which only allows 128KiB blobs,
+// allows blobs of any power-of-2 size between 32B and 16MiB (currently).
+//
+// There are 2 facets to data availability:
+// 1. Local (chunks) availability: validator attests to having received and being able to serve its chunks
+// 2. Global (blob) availability: validator attests to the entire blob being available in the network.
+//
+// Because of the sharded nature of EigenDA, each validator only receives a subset of the blob's content.
+// In order to attest to global availability, it thus needs to know how many chunks there are in total,
+// and to make sure that the chunks it receives are actually proportional to its stake. This is why
+// BlobCommitments contains a length field, as well as a proof of this length (LengthCommitment + LengthProof).
+//
+// Here's an example scenario which shows that EigenDA could go wrong without this Length.
+// In the extreme case, a malicious disperser could just tell the validators that the blob size is 1,
+// and ask all validators except for one to sign off on the commitment. For a slightly more involved but
+// analogous scenario, assume a network of 8 DA nodes with uniform stake distribution, and coding ratio 1/2.
+// For a blob containing 128 field elements (FEs), each node gets 128*2/8=32 FEs, meaning that any 4 nodes can
+// join forces and reconstruct the data. Now assume a world without length proof; a malicious disperser colluding
+// with a client disperses the same blob/commitment, but claims that the blob only has length of 4 FEs.
+// He sends each node 4*2/8=1 FE. The chunks submitted to the nodes match the commitment, so the nodes accept
+// and sign over the blob’s batch. But now there are only 8 FEs in the system, which is not enough to reconstruct
+// the original blob (need at least 128 for that).
+//
+// ----- Length Commitment + Length Proof Explanation -----
+//
+// In theory, proving an upper bound on the actual blob length is very simple (assuming knowledge of pairings),
+// and would require only a LengthCommitment (no LengthProof needed).
+// - G1 and G2: generators of the bn254 curve groups
+// - BL: blob length (power of 2)
+// - BC_G1: blob commitment; [p(x)]_1 := p(s)G1 (this is the same as our [encoding.BlobCommitments].Commitment)
+// - LC_G1: len commitment; q(x) = x^(2^28-BL)*p(x)
```

Comment:
Was this supposed to be an element of G1 so something like:
```
q(x) = x^(2^28-BL)*p(x) 
[q(x)]_1 = q(s)*G1
```
?

Reply:
yes sir: [ef79936](https://github.com/Layr-Labs/eigenda/pull/2171/commits/ef799364190db409ad7d8f840ec01b13ee373689)